cli/internal/helm/action.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package helm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/cli"
+)
+
+const (
+	// timeout is the maximum time given per helm action.
+	timeout = 10 * time.Minute
+)
+
+type applyAction interface {
+	Apply(context.Context) error
+	ReleaseName() string
+	IsAtomic() bool
+}
```

Comment:
This interface (and the structs implementing it) are only used in the helm package, so they should be private.

Reply:
what do you think about extracting an extra `actions` subpkg as suggested in the description?

Reply:
Creating a subpackage sounds like a good idea.
However I'm unsure on how easy/difficult this would be to do while avoiding import cycles.
Maybe implement this as a follow up PR?

Reply:
created a new ticket